### DISCUSSION TITLE
How can I select a shared config from command line?

### DISCUSSION THREAD

**Discussion by jeffrson** [eslint/eslint#14556] [Q&A]
Hi,

I have a (scoped) package with a certain config, and I want to add another config - but how can I make use of it from the commandline?

The other config works, when "extends" pulls it in via package.json. However, `eslint -c` wants to read this config as a file (which doesn't exist of course - apparently it does not resolve the config via installed packages).
Furthermore, when I convert my shared config into a plugin, this again will work via "extends", but `eslint --plugin` tries to find a plugin of the name combined from the actual plugin in the config, instead of the contained configuration directly.

How can I achieve the goal?

Thanks

**Comment by nzakas** [comment]
Can you show the actual commands you are using and the error messages you’re getting?

**Comment by jeffrson** [comment]
Yes, sure! Though I hope I don't get it too complicated.

What I want to achieve is to have a package.json script, that should lint the source with some additional rules. However, these rules should come from extending predefined configs according to
```
"eslintConfig": {
        "extends": ['plugin:@typescript-eslint/recommended-requiring-type-checking']
}
```
from command line. AFAICT, "extends" is not possible then.

I have a package "@scope/eslint-config" containing shared configs which is used up to now as
```
"eslintConfig": {
        "extends": "@scope"
}
```

The package became a "multi config" package, which would work like
```
"eslintConfig": {
        "extends": "@scope/eslint-config/require-types"
}
```
but for me package.json should contain only default (recommended) rules, while those additional rules shall only be used explicitely from commandline.

Unfortunately,
`npx eslint -c @scope/eslint-config/require-types ` 
fails with the message
`Error: Cannot read config file: <currentPath>\@scope\eslint-config\require-types`

So I tried to rewrite my "eslint-config" into "eslint-plugin" cause I expected a bit more flexibility in providing configs from the plugin.

Now, while
```
"eslintConfig": {
        "extends": "plugin:@scope/eslint-plugin/require-types"
}
```
or even
```
"eslintConfig": {
        "extends": "plugin:@scope/require-types"
}
```
works when calling `npx eslint .`, I could not find out how to specify the desired config name from command line as well, since I only can specify the plugin itself (putting a config there, wants me to install this as a package). Neither this

`npx eslint --plugin @scope/require-types src`
nor that
`npx eslint --plugin @scope/eslint-plugin/require-types src`

succeeds:

`The package "@scope/eslint-plugin-require-types" was not found when ...`
or
`The package "@scope/eslint-plugin/require-types" was not found when ...`

**Reply by nzakas** [reply]
Not being able to load a config on the command line looks like a bug. Can you open an issue for that?

**Comment by jeffrson** [comment]
Thanks - done: https://github.com/eslint/eslint/issues/14574

I don't believe it's really a "bug", but rather a missing feature.
